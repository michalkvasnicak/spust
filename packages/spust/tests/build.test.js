// @flow

import { spawnSync } from 'child_process';
import rimraf from 'rimraf';
import { resolve as resolvePath } from 'path';

describe('build script', () => {
  it('works correctly', () => {
    const workDir = resolvePath(__dirname, '../');
    const srcDir = 'fixtures/test';

    // make sure we have compiled sources
    const proc = spawnSync(resolvePath(__dirname, '../bin/spust'), ['build', srcDir], {
      cwd: workDir,
    });

    expect(Buffer.from(proc.stdout).toString('utf8').trim()).toMatchSnapshot();
    expect(Buffer.from(proc.stderr).toString('utf8').trim()).toMatchSnapshot();
    expect(proc.status).toBe(0);
  });
});
