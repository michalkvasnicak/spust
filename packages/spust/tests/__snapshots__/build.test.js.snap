// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build script works correctly 1`] = `"Successfully built. See bundle directory."`;

exports[`build script works correctly 2`] = `""`;
